ci/ray_ci/automation/docker_tags_lib.py

```diff
@@ -0,0 +1,259 @@
+import json
+import subprocess
+from datetime import datetime
+from typing import List, Optional, Callable
+import os
+import requests
+from dateutil import parser
+
+from ci.ray_ci.utils import logger
+
+bazel_workspace_dir = os.environ.get("BUILD_WORKSPACE_DIRECTORY", "")
+
+
+class DockerHubRateLimitException(Exception):
+    """
+    Exception for Docker Hub rate limit exceeded.
+    """
+
+    def __init__(self):
+        super().__init__("429: Rate limit exceeded for Docker Hub.")
+
+
+class RetrieveImageConfigException(Exception):
+    """
+    Exception for failing to retrieve image config.
+    """
+
+    def __init__(self):
+        super().__init__("Failed to retrieve image config.")
+
+
+class AuthTokenException(Exception):
+    """
+    Exception for failing to retrieve auth token.
+    """
+
+    def __init__(self, message: str):
+        super().__init__(f"Failed to retrieve auth token from {message}.")
+
+
+def get_docker_auth_token(namespace: str, repository: str) -> Optional[str]:
+    """
+    Retrieve Docker Registry token.
+
+    Args:
+        namespace: Docker namespace
+        repository: Docker repository
+
+    Returns:
+        Auth token for Docker.
+    """
```

Comment:
just remove the doc here, it is restating the function name.

Reply:
done